internal/gossiper/strategy.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossiper
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+type TargetStrategy[T any] interface {
+	Target(ctx context.Context, txs []T) ([]GossipContainer[T], error)
```

Comment:
We could make it on a per tx basis, but it can be more efficient if we batch handling the assignments.

Reply:
Hmm actually this will be more efficient for DSMR by batching, but doing this one by one will also force us to fetch the proposers for each individual call and potentially get an inconsistent view of what the proposer list will be. I think it is better to continue to allow the targeter to batch tx assignments.